video/transcoder/create_job_from_ad_hoc.go

```diff
@@ -0,0 +1,110 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transcoder
+
+// [START transcoder_create_job_from_ad_hoc]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	transcoder "cloud.google.com/go/video/transcoder/apiv1beta1"
+	transcoderpb "google.golang.org/genproto/googleapis/cloud/video/transcoder/v1beta1"
+)
+
+// createJobFromAdHoc creates a job based on a given configuration. See
+// https://cloud.google.com/transcoder/docs/how-to/jobs#create_jobs_ad_hoc
+// for more information.
+func createJobFromAdHoc(w io.Writer, projectID string, location string, inputUri string, outputUri string) error {
+	// projectID := fmt.Sprintf("my-project-id")
+	// location := fmt.Sprintf("us-central1")
+	// inputUri := fmt.Sprintf("gs://my-bucket/my-video-file")
+	// outputUri := fmt.Sprintf("gs://my-bucket/my-output-folder")
+	ctx := context.Background()
+	client, err := transcoder.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewClient: %v", err)
+	}
+	defer client.Close()
+
+	req := &transcoderpb.CreateJobRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/%s", projectID, location),
+		Job: &transcoderpb.Job{
+			InputUri:  inputUri,
+			OutputUri: outputUri,
+			JobConfig: &transcoderpb.Job_Config{
+				Config: &transcoderpb.JobConfig{
+					ElementaryStreams: []*transcoderpb.ElementaryStream{
+						&transcoderpb.ElementaryStream{
+							Key: "video_stream0",
+							ElementaryStream: &transcoderpb.ElementaryStream_VideoStream{
+								VideoStream: &transcoderpb.VideoStream{
+									Codec:        "h264",
+									BitrateBps:   550000,
+									FrameRate:    60,
+									HeightPixels: 360,
+									WidthPixels:  640,
+								},
+							},
+						},
+						&transcoderpb.ElementaryStream{
+							Key: "video_stream1",
+							ElementaryStream: &transcoderpb.ElementaryStream_VideoStream{
+								VideoStream: &transcoderpb.VideoStream{
+									Codec:        "h264",
+									BitrateBps:   2500000,
+									FrameRate:    60,
+									HeightPixels: 720,
+									WidthPixels:  1280,
+								},
+							},
+						},
+						&transcoderpb.ElementaryStream{
+							Key: "audio_stream0",
+							ElementaryStream: &transcoderpb.ElementaryStream_AudioStream{
+								AudioStream: &transcoderpb.AudioStream{
+									Codec:      "aac",
+									BitrateBps: 64000,
+								},
+							},
+						},
+					},
+					MuxStreams: []*transcoderpb.MuxStream{
+						&transcoderpb.MuxStream{
+							Key:               "sd",
+							Container:         "mp4",
+							ElementaryStreams: []string{"video_stream0", "audio_stream0"},
+						},
+						&transcoderpb.MuxStream{
+							Key:               "hd",
+							Container:         "mp4",
+							ElementaryStreams: []string{"video_stream1", "audio_stream0"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	response, err := client.CreateJob(ctx, req)
```

Comment:
Should this show how to wait for the job, even if only in a comment?

Reply:
Added comment.